### PR TITLE
CONTRIBUTING: add more hints on what is required to cut a release

### LINKED ISSUE #1991: Improve release docs in CONTRIBUTING
**Describe the bug**: 

We can improve the release process documentation a bit:

- add that the PR body should be empty?
- add more details on creating the release in GH, note that we should substitute the changelog with what we added to the tag (git tag -v vX.y.z).
- sort out how vX.x branch is updated since it's protected 

**To Reproduce**

1. Follow CONTRIBUTING instructions to cut a release

**Additional context**

First batch of changes here https://github.com/elastic/apm-agent-python/pull/1989

### PR DESCRIPTION
**Pull request by xrmx**
## What does this pull request do?

Add more hints on how to release a new version of the agent

## Related issues

Closes #1991

### CODE REVIEW

**Review by xrmx — COMMENTED** [review]
Need more hints

https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key

**Comment by xrmx** [comment]
One also need to add some cautions when cloning.

```
ERROR: The 'elastic' organization has enabled or enforced SAML SSO.
To access this repository, you must use the HTTPS remote with a personal access token or SSH with an SSH key and passphrase that has been authorized for this organization.
Visit https://docs.github.com/articles/authenticating-to-a-github-organization-with-saml-single-sign-on/ for more information.
```

**Comment by xrmx** [comment]
Cannot push to 6.x branch because it's protected:

```
$ git push upstream 6.x
0 oggetti totali (0 delta), 0 riutilizzati (0 delta), 0 riutilizzati nel file pack
remote: error: GH006: Protected branch update failed for refs/heads/6.x.
remote: error: 3 of 4 required status checks have not succeeded: 2 expected and 1 failing.
To github.com:elastic/apm-agent-python.git
 ! [remote rejected]   6.x -> 6.x (protected branch hook declined)
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -174,6 +174,10 @@ should "Squash and merge".
 
 ### Releasing
 
+Releases tags are signed so you need to have a PGP key set up, you can follow Github documentation on [creating a key](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) and
```
Comment (xrmx) [inline CONTRIBUTING.md:177]: Well you can sign with the ssh keys too, that would simplify things

Comment (xrmx) [reply CONTRIBUTING.md:177]: And my tags are not verified :sweat:

**Comment by xrmx** [comment]
Updated after the discussion

**Review by basepi — APPROVED** [review]
This looks great!

**Pull request merged**